### ISSUE TITLE
a YAML::NewLine at the end of a sequence brake the next node

### ISSUE THREAD

**Issue by Raffaello** [jbeder/yaml-cpp#902]
eg:

having a sort of this code snippet:

```
out << YAML::Key << "k" << YAML::Value
					<< YAML::Flow << YAML::BeginSeq;
				for (int i = 0; i < 2; i++) {
					out << YAML::BeginMap
						<< YAML::Key << "i" << YAML::Value << i
						<< YAML::EndMap 
						<< YAML::Newline
						;
				}
				out << YAML::EndSeq;
```

if that node is running through another outer loop of Begin/End Map is generating a sort of this result:
```yml
nodeA:   
    k: [{i: 0},
 {i:1},
  ]NodeB:
    k: [{i: 0},
 {i:1},
  ]

```
instead without the `YAML::NewLine` is:
```yml
nodeA:  
    k: [{i: 0}, {i:1}]
  NodeB:
    k: [{i: 0}, {i:1}]
```

it seems that the last new line before ending the sequence should be "ignored" and is not, breaking the next node alignment and resulting in an invalid YAML file.

of course there is a workaround in the code, and just avoid the last new line to be emitted.

but is very not elegant and require to do n-1 in the loop and repeat the inside loop block without the newline,
a better solution would be nice if the `yaml-cpp` library handle that edge case, basically removing the newline (or ignoring it) when the `EndSeq` is emitted.

---

**Comment by Raffaello** [comment]
it seems that the new line change a global state and not just the child state of that specific sequence:
https://github.com/jbeder/yaml-cpp/blob/72fe73a1043bef4a4f9e7032132f2aa50865d97e/src/emitter.cpp#L253

and then:
https://github.com/jbeder/yaml-cpp/blob/41001d1bf9c579598d10ce7b2a3552bebe8a7155/src/emitterstate.cpp#L56

and then when i start the "nodeB" as a map the method hasBegun return true, when instead should return false in my perspective at least:

https://github.com/jbeder/yaml-cpp/blob/0d5c57150cebb430fcc216d8066dbb28aeaafe8e/src/emitterstate.h#L69-L71

**Comment by Raffaello** [comment]
it seems alsto the method `IndentTo(indent);` could potentially have an assert:
`assert  `.col()` <= indent` not sure if it is that intention or can just have an higher col position than the indent.

**Comment by jbeder** [comment]
Yes, this is definitely a bug, but it's not obvious how to fix it. Open to
suggestions.

**Comment by jbeder** [comment]
Ah, I see you submitted a PR. Reviewed.

**Comment by jbeder** [comment]
Fixed by #903
